libs/s25main/mapGenerator/Random.h

```diff
@@ -22,14 +22,19 @@
 #include "mapGenerator/MapSettings.h"
 #include "mapGenerator/Map.h"
 
+namespace rttr {
+namespace mapGenerator {
+
 class Random
 {
 private:
-    RandomUtility rnd_;
+    algorithm::RandomUtility rnd_;
 
 public:
-    Random(RandomUtility& rnd) : rnd_(rnd) {}
+    Random(algorithm::RandomUtility& rnd) : rnd_(rnd) {}
     Map* Create(const MapSettings& settings);
```

Comment:
This is a single method of a class with a generic name `Random`. I know we programmers are bad with names but of a class `Random` I'd expect something to create random numbers with...
In this case: Get rid of the class and make it a single method `std::unique_ptr<Map> createRandomMap(RandomUtility rnd, const MapSettings& settings);` I recommend watching https://www.youtube.com/watch?v=WLDT1lDOsb4. The title says it all: "Free your functions!" ;)
This is a good example for a free function, because how is it used? You create 1 instance, then call Create once and then destroy the class, right? Why the need to store state then? If you expect the state to be kept for some reason, then go for a class. But maybe name it `RandomMapCreator` or so.

Reply:
Absolutely no need to keep the state. Since I'm just in the process of removing some (pretty useless) classes, I created different functions instead. 

Also the name Random was stupid indeed. Originally the name came from the fact that I named all the classes for the different random map styles according to their enum value. 

I remember you had another note on the random map styles to merge them into a single function which just takes parameters (like lake size, ...). I leans against the random map generation in Age of Empires where the user can just select different map styles rather than having to deal with lots of parameters, I quite like the idea. However, it might be nice to have the parameters available, I see the point as well, but I would rather prefer to leave it like this for now (as it's already usable and produces good results) and focus on the other review changes.

Reply:
I tried to improve this a bit by removing duplicate code.